openmoc/materialize.py

```diff
@@ -2,170 +2,356 @@
+                                 domain_type, domain_spec)
+
+        # Search for optional cross sections
+        if 'fission' in domain_group:
+            sigma = _get_numpy_array(domain_group, 'fission', suffix)
+            material.setSigmaF(sigma)
+            py_printf('INFO', 'Loaded "fission" MGXS for "%s %s"', 
+                      domain_type, domain_spec)
+
+    # Return collection of materials
+    return materials
 
-      if 'Total XS' in data[name].keys():
-        new_material.setSigmaT(data[name]['Total XS'])
 
-      if 'Scattering XS' in data[name].keys():
-        new_material.setSigmaS(data[name]['Scattering XS'])
-
-      if 'Fission XS' in data[name].keys():
-        new_material.setSigmaF(data[name]['Fission XS'])
-
-      if 'Nu Fission XS' in data[name].keys():
-        new_material.setNuSigmaF(data[name]['Nu Fission XS'])
-
-      if 'Chi' in data[name].keys():
-        new_material.setChi(data[name]['Chi'])
-
-      # Add this material to the list
-      materials[name] = new_material
-
-
-  ##############################################################################
-  #                      UNSUPPORTED DATA FILE TYPES
-  ##############################################################################
-  else:
-    py_printf('ERROR', 'Unable to materialize using filename %s ' + \
-              'since it has an unkown extension. Supported ' + \
-              'extension types are .hdf5 and .py', filename)
-
-
-
-  # Return the list of materials
-  return materials
+##
+# @brief This routine loads an OpenMC Library of multi-group cross section data.
+# @details The routine instantiates Material objects with multi-group cross
+#          section data and returns a dictionary of each Material object keyed
+#          by its ID. An OpenMOC Geometry may optionally be given and the
+#          routine will directly insert the multi-group cross sections into each
+#          Material in the Geometry. If a Geometry is passed in, Material
+#          objects from the Geometry will be used in place of those instantiated
+#          by this routine.
+# @param mgxs_lib an openmc.mgxs.Library object with cross section data
+# @param geometry an optional geometry populated with materials, cells, etc.
+# @return a dictionary of Material objects keyed by ID
+def load_openmc_mgxs_lib(mgxs_lib, geometry=None):
+
+    # Attempt to import openmc
+    try:
+        import openmc
+    except ImportError:
+        py_printf('ERROR', 'The OpenMC code must be installed on your system')
+
+    if not isinstance(mgxs_lib, openmc.mgxs.Library):
+        py_printf('ERROR', 'Unable to load cross sections from %s which is not '
+                           'an openmc.mgxs.Library object', str(type(mgxs_lib)))
+
+    if geometry and 'openmoc.Geometry' not in str(type(geometry)):
+        py_printf('ERROR', 'Unable to load cross sections for "%s" which '
+                           'is not an OpenMOC Geometry', str(type(geometry)))
+
```

Comment:
It looks like there is a lot of repeated code from the `load_openmc_mgxs_lib()` and `load_from_hdf()` functions. Could a helper function be used to include the common code blocks in one location?